packages/client/src/upload.ts

```diff
@@ -129,6 +128,11 @@ export async function* upload(
 
         let err;
         let result;
+        const abortController = new AbortController();
+        abortControllers.add(abortController);
+        if (aborted) {
+          abortController.abort();
```

Comment:
Good question. I did this to match what happened before when there was just 1 abort controller. All file uploads were queued up, and when one file upload errored then new uploads would fetch with the already aborted `abortController` (effectively cancelling them). One upload erroring doesn't automatically cancel new uploads from starting because they are all queued in a [`forEach`](https://github.com/vercel/vercel/pull/12647/files#diff-38a24c9ef933a33a4a249552ba0411de3aa9ade727d65875c36fa2df7da466e3R88).

This worked, but now I'm realizing now each remaining upload did some unnecessary work between the `await semaphore.acquire();` and `fetch`.

Changed this to `return bail()` instead as early as possible, and manually tested it!